src/common/actions/name-ownership.js

```diff
@@ -0,0 +1,88 @@
+import 'isomorphic-fetch';
+
+import { getError } from '../helpers/api';
+import { conf } from '../helpers/config';
+
+import { requestRegisterName, getPackageUploadRequestMacaroon } from './register-name';
+
+// action types
+export const CHECK_NAME_OWNERSHIP_REQUEST = 'CHECK_NAME_OWNERSHIP_REQUEST';
+export const CHECK_NAME_OWNERSHIP_SUCCESS = 'CHECK_NAME_OWNERSHIP_SUCCESS';
+export const CHECK_NAME_OWNERSHIP_ERROR = 'CHECK_NAME_OWNERSHIP_ERROR';
+
+// name ownership statuses
+export const NAME_OWNERSHIP_NOT_REGISTERED = 'NAME_OWNERSHIP_NOT_REGISTERED';
+export const NAME_OWNERSHIP_ALREADY_OWNED = 'NAME_OWNERSHIP_ALREADY_OWNED';
+export const NAME_OWNERSHIP_REGISTERED_BY_OTHER_USER = 'NAME_OWNERSHIP_REGISTERED_BY_OTHER_USER';
+
+export async function internalNameOwnership(root, discharge, snapName) {
```

Comment:
Moved it out of `actions/register-name.js`, as it starts to make more sense to keep it separately.